riscv/src/reachability.rs

```diff
@@ -82,7 +86,8 @@ fn extract_replacements(statements: &[Statement]) -> BTreeMap<&str, &str> {
         .iter()
         .filter_map(|s| match s {
             Statement::Directive(dir, args) if dir.as_str() == ".set" => {
-                if let [Argument::Symbol(from), Argument::Symbol(to)] = &args[..] {
```

Comment:
This could maybe profit from a function
```
fn try_as_symbol(e: &Argument) -> Option<&String> {
  if let Argument::Expression(Expresison::Symbol(s)) = &e { Some(s) } else { None }
}
```

Reply:
Just tried and looks horrible. Best to just spell out the type names.
```rs
                if let [Some(from), Some(to)] = TryInto::<[Option<&str>; 2]>::try_into(
                    args[..].iter().map(try_as_symbol).collect::<Vec<_>>(),
                )
                .unwrap()
```